integration/data.go

```diff
@@ -0,0 +1,318 @@
+// Copyright (c) 2016 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package integration
+
+import (
+	"fmt"
+	"sort"
+	"testing"
+	"time"
+
+	"github.com/m3db/m3aggregator/aggregation"
+	"github.com/m3db/m3aggregator/aggregator"
+	"github.com/m3db/m3metrics/metric"
+	"github.com/m3db/m3metrics/metric/aggregated"
+	"github.com/m3db/m3metrics/metric/unaggregated"
+	"github.com/m3db/m3metrics/policy"
+	"github.com/m3db/m3x/time"
+	"github.com/stretchr/testify/require"
+)
+
+var (
+	testCounterVal        = int64(123)
+	testBatchTimerVals    = []float64{1.5, 2.5, 3.5, 4.5, 5.5}
+	testGaugeVal          = 456.789
+	testVersionedPolicies = policy.VersionedPolicies{
+		Version: 1,
+		Policies: []policy.Policy{
```

Comment:
Do we have any benchmark tests here? It would be good to know what the cost of applying each policy is and what the upper limit is both in terms of number of policies and number of time series per policy.